src/Parser.zig

```diff
@@ -3809,6 +3815,43 @@ fn unExpr(p: *Parser) Error!Result {
     }
 }
 
+fn builtinChooseExpr(p: *Parser) Error!Result {
+    p.tok_i += 1;
+    const l_paren = try p.expectToken(.l_paren);
+    const cond_tok = p.tok_i;
+    var cond = try p.constExpr();
+    if (cond.val == .unavailable) {
+        try p.errTok(.builtin_choose_cond, cond_tok);
+        return error.ParsingFailed;
+    }
+
+    _ = try p.expectToken(.comma);
+
+    var then_expr = if (cond.getBool()) try p.assignExpr() else try p.parseNoEval(assignExpr);
+    try then_expr.expect(p);
+
+    _ = try p.expectToken(.comma);
+
+    var else_expr = if (!cond.getBool()) try p.assignExpr() else try p.parseNoEval(assignExpr);
+    try else_expr.expect(p);
+
+    try p.expectClosing(l_paren, .r_paren);
+
```

Comment:
```suggestion
    try cond.saveValue(p);
```

Reply:
Both my comments turned out to be wrong as `constExpr` already calls `saveValue`. Sorry about the extra work.